### ISSUE TITLE
Accessing Connected Call blocks and never returns.

### ISSUE THREAD

**Issue by bryanrite** [signalwire/signalwire-ruby-old#29]
```ruby
require "signalwire"

# Signalwire::Logger.logger.level = ::Logger::DEBUG

MY_NUMBER = "+155512345657"

class MyConsumer < Signalwire::Relay::Consumer
  contexts ['incoming']
  def on_incoming_call(call)
    call.answer
    call.play_tts 'Welcome to Relay'
    dial = call.connect [[{ type: 'phone', params: { to_number: MY_NUMBER, from_number: call.from, timeout: 30 } }]]
    pp "Connected"
    pp dial.successful
    dial.call.play_tts "Hello!" if dial.successful
    pp "Waiting on Ending"
    dial.call.wait_for_ending
    call.play_tts 'finished. hanging up.'
    call.hangup
  end
end

MyConsumer.new.run
```

If I run the above and dial in, I hear the TTS, then it'll dial my phone number and connect. I can hear myself talking but "Hello!" TTS is never played, nor is "Waiting on Ending" output. When I hang up the connected call, the initial call leg just sits and waits. I have to manually hang it up.

Seems `dial.call.<foo>` stops the execution.

**Comment by bryanrite** [comment]
#30 doesn't seem to fix this. Just tried it again with 2.1.1.

I will see output:
```
"Connected"
true
```
but it just hangs on `dial.call.play_tts "Hello!"`

To reproduce:
* Have a number forward to a relay context.
* Call that number. (Leg 1)
* Leg 1 will get answered, say Welcome to Relay.
* Leg 1 will dial`MY_NUMBER` (Leg2)
* Leg 1 and 2 will be connected, I can hear audio on both.
* `Hello!` is never heard, "Waiting on Ending" is not displayed.
* Hanging up Leg 2 does not trigger `dial.call.wait_for_ending`. Leg 1 continues to hang until I manually hang it up.

**Comment by bryanrite** [comment]
Fixed in #34
